source/cellular_bg96_api.c

```diff
@@ -3204,3 +3381,1175 @@ CellularError_t Cellular_Init( CellularHandle_t * pCellularHandle,
+    CellularAtReq_t atReqNoResponse =
+    {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    /* Make sure the library is open. */
+    cellularStatus = _Cellular_CheckLibraryStatus( pContext );
+
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        LogError( ( "Cellular_MQTTDisconnect: _Cellular_CheckLibraryStatus failed." ) );
+    }
+
+    if( cellularStatus == CELLULAR_SUCCESS )
+    {
+        /* The return value of snprintf is not used.
+         * The max length of the string is fixed and checked offline. */
+        /* coverity[misra_c_2012_rule_21_6_violation]. */
+        ( void ) snprintf(cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE, "AT+QMTDISC=%d", mqttContextId);
+        pktStatus = _Cellular_TimeoutAtcmdRequestWithCallback( pContext, atReqNoResponse, SOCKET_DISCONNECT_PACKET_REQ_TIMEOUT_MS );
+        cellularStatus = _Cellular_TranslatePktStatus( pktStatus );
+    }
+
+    return cellularStatus;
+}
+
+CellularError_t Cellular_MqttPublish(CellularHandle_t cellularHandle,
+                                     uint8_t mqttContextId,
+                                     uint16_t messageId,
+                                     uint8_t qos,
+                                     bool retain,
+                                     const char * topic,
+                                     const char * message,
+                                     uint32_t messageLength,
+                                     /* coverity[misra_c_2012_rule_8_13_violation] */
+                                     uint32_t * sentDataLength )
+{
+    CellularContext_t * pContext = ( CellularContext_t * ) cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    uint32_t sendTimeout = DATA_SEND_TIMEOUT_MS;
+    char cmdBuf[ 6*CELLULAR_AT_CMD_TYPICAL_MAX_SIZE ] = { '\0' };
+    CellularAtReq_t atReqSocketSend =
+    {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+    CellularAtDataReq_t atDataReqMqttPublish =
+    {
+        (uint8_t*)message,
+        messageLength,
+        sentDataLength,
+        NULL,
+        0
+    };
+
+    /* pContext is checked in _Cellular_CheckLibraryStatus function. */
+    cellularStatus = _Cellular_CheckLibraryStatus( pContext );
+
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        LogError( ( "_Cellular_CheckLibraryStatus failed." ) );
+    }
+    else if( ( message == NULL ) || ( sentDataLength == NULL ) || ( messageLength == 0U ) )
+    {
+        LogError( ( "Cellular_MqttPublish: Invalid parameter." ) );
+        cellularStatus = CELLULAR_BAD_PARAMETER;
+    }
+    else
+    {
+        /* Send data length check. */
+        if( messageLength > ( uint32_t ) CELLULAR_MQTT_MAX_SEND_DATA_LEN )
+        {
+            atDataReqMqttPublish.dataLen = ( uint32_t ) CELLULAR_MQTT_MAX_SEND_DATA_LEN;

```

Comment:
Does that mean that when the data is too long that it will just send the maximum it can? Doesn't that mean we'll just send invalid JSON?

Reply:
Error out instead of sending maximum data